pkg/cmd/grafana-server/commands/cli.go

```
@@ -64,43 +65,49 @@ func RunServer(opts standalone.BuildInfo, cli *cli.Context) error {
 	logger := log.New("cli")
 	defer func() {
 		if err := log.Close(); err != nil {
-			fmt.Fprintf(os.Stderr, "Failed to close log: %s\n", err)
+			logger.Error("Failed to close log", "error", err)
```

If we reach this line it's because there is something wrong with the logger that has caused `log.Close()` to fail, which is most likely why the current code uses `fmt.Fprintf` vs trying to use `logger`

---

Yes, exactly. If log.Close() fails, the logger might already be broken, so using fmt.Fprintf(os.Stderr, ...) ensures we can report the error reliably without depending on the (potentially broken) logging infrastructure.